site-src/geps/gep-726.md

```diff
@@ -0,0 +1,266 @@
+type HTTPRequestRedirect struct {
+  // Path defines a path redirect.
+  //
+  // Support: Extended
+  //
+  // +optional
+  Path *HTTPPathRedirect `json:"path,omitempty"`
+  // ...
+}
+
+// HTTPPathRedirectType defines the type of path redirect.
+type HTTPPathRedirectType string
+
+const (
+  // This type of redirect indicates that the complete path will be replaced by
+  // the path redirect value.
+  AbsoluteHTTPPathRedirect HTTPPathRedirectType = "Absolute"
+
+  // This type of redirect indicates that the prefix match will be replaced by
+  // the path redirect value. This type only applies to path matches of type
+  // Prefix. If a Route has other match types, this redirect MUST NOT be applied
+  // to them.
+  PrefixHTTPPathRedirect HTTPPathRedirectType = "Prefix"
+)
+
+// HTTPPathRedirect defines configuration for path redirects.
+type HTTPPathRedirect struct {
+  // Type defines the type of path redirect.
+  //
+  // +kubebuilder:validation:Enum=Absolute;Prefix
+  Type HTTPPathRedirectType `json:"type"`
+
+  // Value defines the HTTP path to redirect to.
+  //
+  // +kubebuilder:validation:MaxLength=1024
+  Value HTTPPathRedirectType `json:"value"`
+}
+```
+
+### Rewrites
+
+A new `RequestRewrite` filter can be added to support rewrites:
+
+```yaml
+kind: HTTPRoute
+apiVersion: gateway.networking.k8s.io/v1alpha2
+metadata:
+  name: http-filter-1
+spec:
+  rules:
+    - matches:
+      - path:
+          type: Prefix
+          value: /
+      filters:
+      - type: RequestRewrite
+        requestRewrite:
+          hostname: foo.com
+          path:
+            type: Prefix
+            value: /bar
+```
+
+This would be represent with the following API additions:
+```go
+// HTTPRequestRewrite defines a filter that modifies a request during
+// forwarding. At most one of these filters may be used on a Route rule. This
+// may not be used on the same Route rule as a HTTPRequestRedirect filter.
+type HTTPRequestRewrite struct {
+  // Hostname is the value to be used to replace the Host header value during
+  // forwarding.
+  //
+  // Support: Extended
+  //
+  // +optional
+  // +kubebuilder:validation:MaxLength=255
+  Hostname *string `json:"hostname,omitempty"`
+
+  // Path defines a path rewrite.
+  //
+  // Support: Extended
+  //
+  // +optional
+  Path *HTTPPathRewrite `json:"path,omitempty"`
+}
+
+// HTTPPathRewriteType defines the type of path rewrite.
+type HTTPPathRewriteType string
+
+const (
+  // This type of rewrite indicates that the prefix match will be replaced by
+  // the path rewrite value. This type only applies to path matches of type
+  // Prefix. If a Route has other match types, this rewrite MUST NOT be applied
+  // to them.
+  PrefixHTTPPathRewrite HTTPPathRewriteType = "Prefix"
+)
+
+// HTTPPathRewrite defines configuration for path rewrites.
+type HTTPPathRewrite struct {
+  // Type defines the type of path rewrite.
+  //
+  // +kubebuilder:validation:Enum=Prefix
+  Type HTTPPathRewriteType `json:"type"`
```

Comment:
Does `Absolute` make sense here too? ie. rewrite `/app/foo/any/thing` to `/new/path`?

Reply:
I think it would, it just wasn't quite as broadly supported. I guess any implementation that supports regex would also be able to accomplish this though.

Reply:
Added absolute as an option in both places as part of transition to shared types.